### PR TITLE
daylight_fao56

### LINKED ISSUE #104: [GA-1516] Meteorology - daylight_fao56
`daylight_fao56` NCL and geocat function

- [ncl_entry for `daylight_fao56`](https://www.ncl.ucar.edu/Document/Functions/Built-in/dewtemp_trh.shtml)
- geocat-comp: [`max_daylight`](https://geocat-comp.readthedocs.io/en/v2024.04.0/user_api/generated/geocat.comp.meteorology.max_daylight.html)
- NCL receipt to `daylight_fao56` and `max_daylight`

### PR DESCRIPTION
**Pull request by cyschneck**
## PR Summary
Add Meteorology `max_daylight` geocat function to resolve NCL `daylight_fao56` function

Covers:
- [daylight_fao56](https://www.ncl.ucar.edu/Document/Functions/Built-in/dewtemp_trh.shtml)

## Related Tickets & Documents
Closes #104

## PR Checklist
**General**
- [x] PR includes a summary of changes
- [x] Link relevant issues, make one if none exist
- If adding a new page, select which type:
  - [x] NCL Entry
  - [x] Receipt

### CODE REVIEW

**Comment by cyschneck** [comment]
Combined `dewtemp` and `daylight` functions into a single NCL entry, which will eventually include additional meteorology functions

**Comment by jukent** [comment]
@cyschneck I won't be able to look at this until Monday.

**Review by kafitzgerald — COMMENTED** [review]
A few suggestions, but overall this looks good. 

The receipt is a little hard to read, but I think a good bit of that is from the formatting which is probably inherited from other receipts and refactoring that is beyond the scope of this PR.

File: ncl/ncl_entries/meteorology.ipynb [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Python Resources\n",
+    "- [GeoCAT-comp dewtemp Documentation](https://geocat-comp.readthedocs.io/en/latest/user_api/generated/geocat.comp.meteorology.dewtemp.html)\n",
+    "- [Convert between different temperature scales in SciPy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.constants.convert_temperature.html)\n",
+    "- [GeoCAT-comp daylight Documentation](https://www.ncl.ucar.edu/Document/Functions/Crop/daylight_fao56.shtml)\n",
```
[the remark this answers is not in the document]
Comment (cyschneck) [reply ncl/ncl_entries/meteorology.ipynb:180]: Thanks, updating it now

**Review by jukent — APPROVED** [review]
Approved, pending @kafitzgerald 's suggestions.

I want to make sure we're remembering all of the formatting and other comments we've made in PRs that are outside the scope of those PRs

**Comment by kafitzgerald** [comment]
The pre-commit failure here looks like it's the same as on https://github.com/NCAR/geocat-examples/pull/618.  

I ended up just commenting docformatter out for now in the pre-commit config and logging an issue to add it back once the new release is out to address this there.

**Comment by cyschneck** [comment]
> logging

Will do

**Review by kafitzgerald — APPROVED** [review]

**Pull request merged**